api/end_device.proto

```diff
@@ -51,7 +51,7 @@ message Session {
   // Time when the session started. Network Server only.
   google.protobuf.Timestamp started_at = 8 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
   // Queued Application downlink messages. Stored in Application Server and Network Server.
-  repeated ApplicationDownlink queued_application_downlinks = 9;
+  repeated ApplicationDownlink queued_application_downlinks = 9 [(validate.rules).repeated.max_items = 100000];
```

Comment:
Don't set the limit here if it can be configured in NS to a potentially higher value that would not be effective.

That also really raises the question what the need is for `uint64`.

Set it in the request so that you cannot enqueue more than N downlink messages at once. Clients can adapt for that.

Reply:
Yeah, that was supposed to be in the request - wrong message